src/renderer/features/network/NetworksList/lib/networks-list-utils.ts

```diff
@@ -0,0 +1,37 @@
+import { ExtendedChain, networkUtils } from '@entities/network';
+import { Chain, ChainId, Connection, ConnectionStatus } from '@shared/core';
+
+export const networksListUtils = {
+  getExtendedChain,
+  getMetrics,
+};
+
+function getExtendedChain(
+  chains: Chain[],
+  connections: Record<ChainId, Connection>,
+  connectionStatuses: Record<ChainId, ConnectionStatus>,
+): ExtendedChain[] {
+  return chains.map((chain) => {
+    return {
+      ...chain,
+      connection: connections[chain.chainId],
+      connectionStatus: connectionStatuses[chain.chainId],
+    };
+  }, []);
+}
+
+type Metrics = Record<'success' | 'connecting' | 'error', number>;
+function getMetrics(networkList: ExtendedChain[]): Metrics {
```

Comment:
Are you sure that `Metrics` is a good name? Maybe add smth specific for it? `getStatusMetrics` or smth else

Reply:
`getStatusMetrics` sounds better I agree, I'll change